Move Compute Resources topic to Concepts.

steveperry-53 opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/2410)
<!-- Reviewable:end -->

chenopis commented on docs/concepts/configuration/manage-compute-resources-container.md:
```diff
@@ -0,0 +1,424 @@
+
+Specifying resource requests and limits is optional. In some clusters, unset
+limits or requests might be replaced with default values when a Pod is created
+or updated. The default values depends on how the cluster is configured. If the
```
depends -> depend; "The default values **depend** on..."

steveperry-53 replied:
Done.

chenopis commented on docs/concepts/configuration/manage-compute-resources-container.md:
```diff
@@ -0,0 +1,424 @@
+is convenient to talk about Pod resource requests and limits. A
+*Pod resource request/limit* for a particular resource type is the sum of the
+resource requests/limits of that type for each Container in the Pod, with
+unset values treated as zero (or equal to default values in some cluster
```
What type of cluster configurations would lead to unset values being set to default values instead of being treated as zero? These are quite different outcomes, right? Is there somewhere else that can be linked to where this is described in more detail?

steveperry-53 replied:
@pwittrock, Can you comment on this?

chenopis commented on docs/concepts/configuration/manage-compute-resources-container.md:
```diff
@@ -0,0 +1,424 @@
+
+## Meaning of CPU
+
+Limits and requests for CPU are measured in cpus.
```
Missing 'resources'? Also, since 'cpus' is ambiguously plural, consider using italics and 'units' to clarify, i.e. "Limits and requests for CPU **resources** are measured in _cpu_ units."

steveperry-53 replied:
Done.

chenopis commented on docs/concepts/configuration/manage-compute-resources-container.md:
```diff
@@ -0,0 +1,424 @@
+is configured for your cluster, then Pod resource usage can be retrieved from
+the monitoring system.
+
+## Troubleshooting
```
Just wondering in general, is a Concept page the proper place where we should have the associated Troubleshooting section?

steveperry-53 replied:
This is OK for now. We can fine tune later if we have time.

chenopis replied:
Ok, sounds good. We can discuss this later.

pwittrock commented:
@vish Is this something the Node team should review?

vishh commented:
Yeah, I can review this.

steveperry-53 commented:
@vishh, ping

vishh commented on docs/concepts/configuration/manage-compute-resources-container.md:
```diff
@@ -0,0 +1,424 @@
+If a Container exceeds its memory limit, it might be terminated. If it is
+restartable, the kubelet will restart it, as with any other type of runtime
+failure.
+
```
Add: `If a Container exceeds its memory request, it is likely that its pod might be evicted whenever the node runs out of memory`.

steveperry-53 replied:
Done.

vishh commented on docs/concepts/configuration/manage-compute-resources-container.md:
```diff
@@ -0,0 +1,424 @@
+In the preceding example, you can see from the `Allocated resources` section
+that if a Pod asks for more than 90 millicpus or more than 1341MiB of memory, it
+will not fit on the node.
+
```
Introduce [`NodeAllocatable`](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node-allocatable.md) here

steveperry-53 replied:
Done.

vishh commented on docs/concepts/configuration/manage-compute-resources-container.md:
```diff
@@ -0,0 +1,424 @@
+You can see that the Container was terminated because of `reason:OOM Killed`,
+where `OOM` stands for Out Of Memory.
+
+## Opaque integer resources (Alpha feature)
```
I'd avoid including alpha features here. How about linking to them instead?

steveperry-53 replied:
I like the idea of not putting alpha features on the same page with the GA/Beta content. But I'd like to leave this as-is for now. I will discuss this with the doc team and others who are interested. We need to come up with a site-wide policy for talking about alpha features.

vishh replied:
sgtm

steveperry-53 commented:
@vishh, I addressed your comments in https://github.com/kubernetes/kubernetes.github.io/pull/2410/commits/2c219b87583abb0651bfccf9b040ce02a06a2e27. I also replied to some of your comments in line.

steveperry-53 commented on docs/concepts/configuration/manage-compute-resources-container.md:
```diff
@@ -0,0 +1,424 @@
+## How Pods with resource limits are run
+
+When the kubelet starts a Container of a Pod, it passes the CPU and memory limits
+to the container runner: Docker or rkt.
```
Done.

steveperry-53 commented:
@vishh, Could you take a look at this statement to see if it's accurate?

"In the preceding example, you can see from the Allocated resources section that if a Pod asks for more than 90 millicpus or more than 1341MiB of memory, it will not fit on the node."

 I understand the 90, but I don't understand the 1341.

vishh commented on docs/concepts/configuration/manage-compute-resources-container.md:
````diff
@@ -0,0 +1,425 @@
+`kubectl describe nodes` command. For example:
+
+```shell
+$ kubectl describe nodes gke-cluster-4-386701dd-node-ww4p
````
This needs to be updated to include `Allocatable` in addition to `Allocated`. You can use the snippet below and update the text to use the number from this snippet.

```text
$ kubectl.sh describe nodes e2e-test-minion-group-4lw4
Name:			e2e-test-minion-group-4lw4
[ ... lines removed for clarity ...]
Capacity:
 alpha.kubernetes.io/nvidia-gpu:	0
 cpu:					2
 memory:				7679792Ki
 pods:					110
Allocatable:
 alpha.kubernetes.io/nvidia-gpu:	0
 cpu:					1800m
 memory:				7474992Ki
 pods:					110
[ ... lines removed for clarity ...]
Non-terminated Pods:		(5 in total)
  Namespace			Name							CPU Requests	CPU Limits	Memory Requests	Memory Limits
  ---------			----							------------	----------	---------------	-------------
  kube-system			fluentd-gcp-v1.38-28bv1					100m (5%)	0 (0%)		200Mi (2%)200Mi (2%)
  kube-system			kube-dns-3297075139-61lj3				260m (13%)	0 (0%)		100Mi (1%)170Mi (2%)
  kube-system			kube-proxy-e2e-test-vishnuk-minion-group-4lw4		100m (5%)	0 (0%)		0 (0%)		0 (0%)
  kube-system			monitoring-influxdb-grafana-v4-z1m12			200m (10%)	200m (10%)	600Mi (8%)600Mi (8%)
  kube-system			node-problem-detector-v0.1-fj7m3			20m (1%)	200m (10%)	20Mi (0%)100Mi (1%)
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  CPU Requests	CPU Limits	Memory Requests	Memory Limits
  ------------	----------	---------------	-------------
  680m (34%)	400m (20%)	920Mi (12%)	1070Mi (14%)
```

Specifically, if a pod requests more than `1120m` CPUs or `6.23Gi`, it will not fit on the node.

vishh commented:
@steveperry-53 I posted a comment that hopefully answers your question. Essentially `Availability` on a node is `Allocatable` - `Allocated` resources

steveperry-53 commented:
@vishh, I adjusted the command example according to your comment. Does this look ready to merge now?
